File: 2294-partition-array-such-that-maximum-difference-is-k/2294-partition-array-such-that-maximum-difference-is-k.cpp
class Solution {
public:
    static int partitionArray(vector<int>& nums, int k) {
        sort(nums.begin(), nums.end());
        int partitions=0, n=nums.size();
        for(int i=0; i<n; ){
            i=upper_bound(nums.begin()+i, nums.end(), nums[i]+k)-nums.begin();
            partitions++;
        }
        return partitions;
    }
};

File: 2294-partition-array-such-that-maximum-difference-is-k/2294-partition-array-such-that-maximum-difference-is-k.java
import java.util.Arrays;

class Solution {
  public int partitionArray(int[] nums, int k) {
    Arrays.sort(nums);
    int ans = 1, l = 0;
    for (int i = 1; i < nums.length; i++) {
      if (nums[i] - nums[l] > k) {
        ans++;
        l = i;
      }
    }
    return ans;
  }
}